Add oss registry support (#1715)
oam-dev/kubevela: design/api/vela-plugin-registry-reference.md

> # vela Plugin Registry Reference

Reviewer: ```suggestion
# Kubevela Plugin Registry Reference
```

Author: done

## registry interface intro

> All `Registry` interface defines are two methods:GetCap and ListCaps. Here "Cap" can represent trait or component definition

Reviewer: ```suggestion
All `Registry` interface definitions have two methods: `GetCap` and `ListCaps`. Here "Cap" can represent trait or component definition
```

Author: done
